### ISSUE TITLE
RN 0.41 includes a bundle file that causes node haste conflict

### ISSUE THREAD

**Issue by zawarudo** [airbnb/lottie-react-native#29]
Following https://github.com/airbnb/lottie-react-native#running-the-example-project

You can check out the example project with the following instructions

1. Clone the repo: `git clone https://github.com/airbnb/lottie-react-native.git`
2. Open: `cd lottie-react-native and Install: npm install`
3. Run `npm start to start the packager.`
4. ... error

node 7.4.0
npm 4.1.2

One liner to reproduce.

`git clone https://github.com/airbnb/lottie-react-native.git lottie-react-native && cd $_; npm install; npm start;`

```
Looking for JS files in
   ./lottie-react-native 

Loading dependency graph...
React packager ready.

Failed to build DependencyGraph: @providesModule naming collision:
  Duplicate module name: deprecated
  Paths: ./lottie-react-native/node_modules/react-native/local-cli/server/middleware/heapCapture/bundle.js collides with ./lottie-react-native/node_modules/react-native/Libraries/Renderer/src/shared/utils/deprecated.js

This error is caused by a @providesModule declaration with the same name across two different files.
Error: @providesModule naming collision:
  Duplicate module name: deprecated
  Paths: ./lottie-react-native/node_modules/react-native/local-cli/server/middleware/heapCapture/bundle.js collides with ./lottie-react-native/node_modules/react-native/Libraries/Renderer/src/shared/utils/deprecated.js

This error is caused by a @providesModule declaration with the same name across two different files.
    at HasteMap._updateHasteMap (./lottie-react-native/node_modules/react-native/packager/react-packager/src/node-haste/DependencyGraph/HasteMap.js:158:13)
    at module.getName.then.name (./lottie-react-native/node_modules/react-native/packager/react-packager/src/node-haste/DependencyGraph/HasteMap.js:119:31)

npm ERR! Darwin 15.6.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "start"
npm ERR! node v7.4.0
npm ERR! npm  v4.1.2
npm ERR! code ELIFECYCLE
npm ERR! lottie-react-native@1.0.1 start: `node node_modules/react-native/local-cli/cli.js start`
npm ERR! Exit status 1

```

**Comment by lelandrichardson** [comment]
Interesting. I think this is an RN 0.41 bug. It looks like there's some bundle file being created that shouldn't be. If you delete that 2nd file that's listed it should work. I'll surface to the RN team and see what they say.

**Comment by grabbou** [comment]
Can you try reproducing on 0.41.0-rc.1? I think that's fixed now.

**Comment by ABleas** [comment]
I have a similar issue that seems to be from conflicts with the Pod dependencies. I'm using a fresh react native 0.41 project, following instructions to add lottie on ios with Pods.

`Loading dependency graph...
React packager ready.

Failed to build DependencyGraph: @providesModule naming collision:
  Duplicate module name: react-native-packager
  Paths: ./ReactNative/LogoTest/node_modules/react-native/packager/package.json collides with ./ReactNative/LogoTest/ios/Pods/React/packager/package.json

This error is caused by a @providesModule declaration with the same name across two different files.
Error: @providesModule naming collision:
  Duplicate module name: react-native-packager
  Paths: ./ReactNative/LogoTest/node_modules/react-native/packager/package.json collides with ./ReactNative/LogoTest/ios/Pods/React/packager/package.json

This error is caused by a @providesModule declaration with the same name across two different files.
    at HasteMap._updateHasteMap (./ReactNative/LogoTest/node_modules/react-native/packager/react-packager/src/node-haste/DependencyGraph/HasteMap.js:158:13)
    at p.getName.then.name (./ReactNative/LogoTest/node_modules/react-native/packager/react-packager/src/node-haste/DependencyGraph/HasteMap.js:133:31)
~
Process terminated. Press <enter> to close the window
`

And I also have 2 build errors in my xcode project I can't make sense of..

`./ReactNative/LogoTest/node_modules/lottie-react-native/lib/ios/LottieReactNative/LottieAnimationViewManager.m:57:37: Incompatible block pointer types sending 'void (^)(RCTUIManager *__strong, NSDictionary<NSNumber *,UIView *> *__strong)' to parameter of type 'RCTViewManagerUIBlock' (aka 'void (^)(RCTUIManager *__strong, RCTSparseArray *__strong)')`

**Comment by zawarudo** [comment]
@grabbou Still broken for me with 0.41.0-rc.1

**Comment by lelandrichardson** [comment]
@ABleas if pods is including a different version of react-native, then I think your project might be set up improperly. Pods should be installing react from node_modules.

**Comment by grifotv** [comment]
deleting the file `./lottie-react-native/node_modules/react-native/Libraries/Renderer/src/shared/utils/deprecated.js` does make it work.

**Comment by lelandrichardson** [comment]
@grifotv deleting the `bundle.js` is more advisable

**Comment by IljaDaderko** [comment]
@lelandrichardson could you please point to exact location where such bundle.js file is located?

**Comment by lelandrichardson** [comment]
@IljaDaderko `$REACT_NATIVE_MODULE_ROOT//local-cli/server/middleware/heapCapture/bundle.js` where `$REACT_NATIVE_MODULE_ROOT` would be `./node_modules/react-native` for most people if they're current working directory is where their package.json is.

**Comment by IljaDaderko** [comment]
@lelandrichardson thanks, and to confirm 0.41.0-rc.1 still doesn't fix this for me either, I believe you have discussion going on over at react-native about this.

**Comment by lelandrichardson** [comment]
This is fixed in 0.41.1
